src/codeu/chat/server/JSON.java

```diff
@@ -68,141 +69,182 @@ public Model readFromFile (String file) throws IOException
             // will be used and returned.
             System.out.println("Invalid file");
         }
-       if (jp!=null) {
+        if (jp!=null) {
             // If the file exists, then the file will be parsed
-           // in this set of code
-           jp.nextToken();
+            // in this set of code
+            jp.nextToken();
             while (jp.nextToken() != JsonToken.END_OBJECT)
             // This ensures that we haven't reached the end of the
             // file because the data is being parsed in such a way
             // that this END_OBJECT specifically relates to the
             // very last } in the json file
             {
-            if(jp.getText().equals("users")) {
-                // If the object type represents a User, then
-                // specific steps will take place to create
-                // a User variable
-                jp.nextToken();
-                while (jp.nextToken() != JsonToken.END_ARRAY) {
+                if(jp.getText().equals("users")) {
+                    // If the object type represents a User, then
+                    // specific steps will take place to create
+                    // a User variable
+                    //jp.nextToken();
+                    Time time = null;
                     String name = "";
                     Uuid id = null;
-                    Time time = null;
-                    jp.nextToken();
-                    if (jp.getText().equals("name")) {
-                        jp.nextToken();
-                        name = jp.getText();
-                    }
-                    jp.nextToken();
-                    if (jp.getText().equals("uuid")) {
-                        jp.nextToken();
-                        try {
-                            id = Uuid.parse(jp.getText());
-                        } catch (IOException e) {
-                            System.out.println("Invalid Uuid");
+                    while (jp.nextToken() != JsonToken.END_ARRAY) {
+                        if (jp.getText().equals("name")) {
+                            jp.nextToken();
+                            name = jp.getText();
+                            //jp.nextToken();
+                        }
+                        if (jp.getText().equals("uuid")) {
```

Comment:
So pretty much the way that I've set it up is that it only goes onto the next tag at the beginning of the while loop. The way that I made order not matter was to not have the parser moving onto the nextToken otherwise. In fact, it no longer will go through that order in one iteration of a while, because even if it was in that order, it would have to go back through the while statement just to move on. jp.nextToken() is the only thing that moves it to to the next value, getText() just returns the value at that token. If you called getText() multiple times, the same piece of text will just just get repeated.